IdentityCore/src/controllers/broker/mac/MSIDXpcInteractiveTokenRequestController.m

```diff
@@ -0,0 +1,125 @@
+                                                 error:error];
+    if (self)
+    {
+        _fallbackController = fallbackController;
+    }
+    
+    return self;
+}
+
+#pragma mark - MSIDRequestControlling
+
+- (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning interactive Xpc service flow.");
+    MSIDInteractiveTokenRequest *request = [self.tokenRequestProvider interactiveXpcTokenRequestWithParameters:self.interactiveRequestParamaters];
+    
+    MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *result, NSError *error)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Interactive broker extension flow finished. Result %@, error: %ld error domain: %@", _PII_NULLIFY(result), (long)error.code, error.domain);
+        if (!error)
+        {
+            /**
+             Throttling service: when an interactive token succeed, we update the last refresh time of the throttling service
+             */
+            [MSIDThrottlingService updateLastRefreshTimeDatasource:request.extendedTokenCache context:self.interactiveRequestParamaters error:nil];
+           
+        }
+        else if ([self shouldFallback:error])
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Falling back to local controller.");
+            
+            [self.fallbackController acquireToken:completionBlock];
+            return;
+        }
+        
+        completionBlock(result, error);
+    };
+    
+    [self acquireTokenWithRequest:request completionBlock:completionBlockWrapper];
+}
+
++ (BOOL)canPerformRequest
+{
+    if (@available(macOS 13, *)) {
+        return [MSIDXpcSingleSignOnProvider canPerformRequest:MSIDXpcProviderCache.sharedInstance];
+    } else {
+        return NO;
+    }
+}
+
+#pragma mark - Private
+
+- (BOOL)shouldFallback:(NSError *)error
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking if we should fallback to fallback controller, error: %ld error domain: %@.", (long)error.code, error.domain);
+    
+    if (!self.fallbackController)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"fallback controller is nil, Xpc service controller should fallback: NO");
+        return NO;
+    }
+    
+    // If it is MSIDErrorDomain and Sso Extension returns unexpected error, we should fall back to local controler and unblock user
+    if (![error.domain isEqualToString:ASAuthorizationErrorDomain] && ![error.domain isEqualToString:MSIDErrorDomain]) return NO;
+    
+    BOOL shouldFallback = NO;
+    switch (error.code)
+    {
+        // TODO: 3236668 define XPC error code that should fallback
+        case ASAuthorizationErrorNotHandled:
+        case ASAuthorizationErrorUnknown:
+        case ASAuthorizationErrorFailed:
+        case MSIDErrorSSOExtensionUnexpectedError:
```

Comment:
We discussed offline about creating the XPC errors to replace these errors later. Checking if it would help to configure the MSIDErrorXPCUnexpectedError now?